data_preparation/predictor_data_prep.r

```diff
@@ -241,93 +247,146 @@ process_ts_raster <- function(region, ind_filename, indicator, crop_mask_filenam
+      if (indicator_source == "AgERA5") {
+        indicator_path = file.path(PREDICTORS_DATA_PATH, indicator_source,
+                                   paste(region, indicator_source, sep="_"), indicator)
+      } else {
+        indicator_path = file.path(PREDICTORS_DATA_PATH, indicator_source, indicator)
+      }
+
+      found_raster_file <- FALSE
+      while(!found_raster_file) {
+        file_list <- list.files(path=indicator_path,
+                                pattern=glob2rx(paste0(filename_pattern, as.character(start_year), "*")),
+                                full.names=TRUE)
+        if (length(file_list) == 0) {
+          warning(paste("Files not found for", start_year))
+          start_year <- start_year + 1 
+        } else {
+          found_raster_file <- TRUE
+        }
+      }
+
+      # Resample crop mask to indicator rastor resolution.
+      ind_rast <- rast(file_list[[1]])
+      crop_mask <- resample(crop_mask, ind_rast, method="bilinear")
+      crop_mask <- crop(crop_mask, region_boundaries)
+      # NOTE: We are overwriting in every iteration (per indicator).
+      # This is fine because resampling is needed per indicator.
+      resampled_filename <- file.path(AGML_ROOT, "crop_masks",
+                                      "resampled_crop_mask.tif")
+      writeRaster(crop_mask, resampled_filename, overwrite=TRUE)
+
+      rm(list=c("ind_rast", "crop_mask"))
+      dfs <- pblapply(start_year:end_year, process_ts_year,
+                      file_path=indicator_path,
+                      filename_pattern=filename_pattern,
+                      indicator=indicator, 
+                      crop_mask_file=resampled_filename,
+                      region_boundaries=region_boundaries, 
+                      cl=num_cpus)
+ 
+      result <- rbindlist(dfs)
+      rm(dfs)
+      gc()
+    } else {
+      ###############
+      # static data #
+      ###############
+
       # NOTE for crop calendars, sos and eos are crop specific.
       if (indicator_source == "ESA_WC_Crop_Calendars") {
-        ind_rast <- rast(file.path(PREDICTORS_DATA_PATH,
-                                   indicator_source,
-                                   paste0(crop, "_", filename_pattern, ".tif")))
+        ind_filename <- file.path(PREDICTORS_DATA_PATH,
+                                  indicator_source,
+                                  paste0(crop, "_", filename_pattern, ".tif"))
       } else {
-        ind_rast <- rast(file.path(PREDICTORS_DATA_PATH,
-                                   indicator_source,
-                                   paste0(filename_pattern, ".tif")))
+        ind_filename <- file.path(PREDICTORS_DATA_PATH,
+                                  indicator_source,
+                                  paste0(filename_pattern, ".tif"))
       }
  
-      # resample
+      # Resample crop mask to the resolution of indicator.
+      ind_rast <- rast(ind_filename)
```

Comment:
Here you can use `win` to skip the crop later

Reply:
Here the idea was to use the whole raster for resampling.